### PR TITLE
Clarify type signature requirement in task 1

### LINKED ISSUE #596: Wrong instruction about makeTreasureChest in part 1 of the description in the learning exercise Treasure factory
In Part 1 of the instructions( just above the Note) this is stated:
"makeTreasureChest should take a List of Chest without specific conditions and return a List of Chest, with the duplicates removed, and the extra condition uniqueTreasure : () added."

But the type signature of `makeTreasureChest` is already given as part of the requirements later and does not have a List as either input or output.

### PR DESCRIPTION
**Pull request by ceddlyburge**
Fixes #596

### CODE REVIEW

**Comment by ceddlyburge** [comment]
@mpizenberg @jiegillet

**Comment by mpizenberg** [comment]
Been away, I’ll check issues and PRs tonight. Thanks.

File: exercises/concept/treasure-factory/.docs/instructions.md [hunk 1]
```diff
@@ -29,12 +29,14 @@ A `Chest` should hold the same data as a `TreasureChest` and should have two typ
 Note that because `Chest` is using a phantom type, and should therefore be opaque to prevent its use anywhere outside of the `TreasureFactory` module.
 In this case, it is not even exposed at all.
 
-`secureChest` should take a `Chest` without specific conditions and return one wrapped in a `Maybe` that has the extra condition `securePassword : ()`.
-`makeTreasureChest` should take a `List` of `Chest` without specific conditions and return a `List` of `Chest`, with the duplicates removed, and the extra condition `uniqueTreasure : ()` added.
+Edit the type signatures of `secureChest` and `uniqueTreasures` to add the constraints using phantom types.
```
Comment (mpizenberg) [inline exercises/concept/treasure-factory/.docs/instructions.md:32]: Could we be a tiny bit more specific by saying "using extensible records as phantom types"?

File: exercises/concept/treasure-factory/.docs/instructions.md [hunk 2]
```diff
@@ -29,12 +29,14 @@ A `Chest` should hold the same data as a `TreasureChest` and should have two typ
...
+`secureChest` should take a `Chest` without specific conditions and return a `Maybe Chest`, with the extra condition `securePassword : ()`.
```
Comment (mpizenberg) [inline exercises/concept/treasure-factory/.docs/instructions.md:33]: And here say something like "with the extra condition `securePassword : ()` in its phantom record".

File: exercises/concept/treasure-factory/.docs/instructions.md [hunk 3]
```diff
@@ -29,12 +29,14 @@ A `Chest` should hold the same data as a `TreasureChest` and should have two typ
...
+`uniqueTreasures` should take a `List Chest` without specific conditions and return a `List Chest`, with the extra condition `uniqueTreasure : ()` added.
```
Comment (mpizenberg) [inline exercises/concept/treasure-factory/.docs/instructions.md:34]: "with the extra condition `uniqueTreasure : ()` added to the phantom record."

**Review by mpizenberg — APPROVED** [review]
Took me a while to grasp what was going on in this exercise as I had not seen it yet ^^.  I think the changes are good. I've given some other options, I'll let you be the judge of if that's useful or not.

**Comment by ceddlyburge** [comment]
Thanks matthieu :)

**Pull request merged**